File: src/main/java/io/gittul/infra/auth/exception/AuthorizationException.java
package io.gittul.infra.auth.exception;

import io.gittul.global.exception.CustomException;
import org.springframework.http.HttpStatus;

public class AuthorizationException extends CustomException {
    public AuthorizationException(String message) {
        super(HttpStatus.FORBIDDEN, message);
    }

    public AuthorizationException() {
        super(HttpStatus.FORBIDDEN, "Authorization Error");
    }

    public static AuthorizationException ADMIN_ONLY = new AuthorizationException("관리자 권한이 필요합니다.");
}

File: app/src/main/kotlin/io/gittul/app/infra/auth/exception/AuthorizationException.kt
package io.gittul.app.infra.auth.exception

import io.gittul.core.domain.user.entity.Role
import io.gittul.core.global.exception.CustomException
import org.springframework.http.HttpStatus

class AuthorizationException : CustomException {
    constructor(message: String?) : super(HttpStatus.FORBIDDEN, message)

    constructor() : super(HttpStatus.FORBIDDEN, "Authorization Error")


    companion object {
        var ADMIN_ONLY: AuthorizationException = AuthorizationException("관리자 권한이 필요합니다.")

        fun requireRole(role: Role): AuthorizationException {
            return AuthorizationException("권한이 없습니다. $role 권한이 필요합니다.")
        }
    }
}
